src/VirtoCommerce.Platform.Core/JsonConverters/PolymorphJsonConverter.cs

```diff
@@ -1,49 +1,108 @@
 using System;
-using System.Linq;
+using System.Collections.Concurrent;
+using System.Reflection;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 using VirtoCommerce.Platform.Core.Common;
-using VirtoCommerce.Platform.Core.DynamicProperties;
-using VirtoCommerce.Platform.Core.Security;
-using VirtoCommerce.Platform.Core.Settings;
 
 namespace VirtoCommerce.Platform.Core.JsonConverters
 {
     public class PolymorphJsonConverter : JsonConverter
     {
-        private static readonly Type[] _knowTypes = { typeof(ObjectSettingEntry), typeof(DynamicProperty), typeof(ApplicationUser), typeof(Role), typeof(PermissionScope) };
+        
+        public class CreateInstanceCacheKey
```

Comment:
Why this code won't work for you? 
```
CacheKey.With(GetType(), type.Name, false);
```
I think this type `CreateInstanceCacheKey` is redundant for this task.

Reply:
Aggree, removed